whyis/database/elasticstore.py

```diff
@@ -0,0 +1,469 @@
+
+    def namespace(self, prefix):
+        return self.__namespace.get(prefix, None)
+
+    def prefix(self, namespace):
+        return self.__prefix.get(namespace, None)
+
+    def namespaces(self):
+        for prefix, namespace in iteritems(self.__namespace):
+            yield prefix, namespace
+
+    def contexts(self, triple=None):
+        assert False, "Not implemented."
+        for i in range(0):
+            yield(URIRef())
+
+    def add_graph(self, graph, refresh=False):
+        self.addN(((s,p,o,graph.identifier) for s,p,o in graph.triples((None, None, None))), refresh=refresh)
+
+    def remove_graph(self, graph):
+        if isinstance(graph, Graph):
+            graph = graph.identifier
+        response = self.session.delete(self.url+"/nanopublication/"+graph.split('/')[-1])
+
+def json_ld_triples(json_response, subject=None, predicate=None, object=None, context=None):
+    #print json_response
+    if 'hits' not in json_response:
+        return
+    for hit in json_response['hits']['hits']:
+        for graph in hit['_source']['graphs']:
+            graphid = rdflib.URIRef(graph['@id'])
+            if context is not None and context != graphid:
+                continue
+            for resource in graph['@graph']:
+                resourceid = rdflib.URIRef(resource['@id'])
+                if resourceid.startswith('_:'):
+                    resourceid = rdflib.BNode(resourceid[2:])
+                else:
+              	    resourceid = rdflib.URIRef(resourceid)
+                if subject is not None and subject != resourceid:
+                    continue
+                predicates = []
+                if predicate is not None:
+                    predicates = [str(predicate)]
+                else:
+                    predicates = list(resource.keys())
+                for pred in predicates:
+                    if pred == '@object' or pred == '@id':
+                        continue
+                    if pred not in resource:
+                        continue
+                    for obj in resource[pred]:
+                        o = None
+                        if '@id' in obj:
+                            id = obj['@id']
+                            if id.startswith('_:'):
+                                o = rdflib.BNode(id[2:])
+                            else:
+                                o = rdflib.URIRef(id)
+                        else:
+                            args = {}
+                            if '@type' in obj:
+                                args['datatype'] = rdflib.URIRef(obj['@type'])
+                            if '@lang' in obj:
+                                args['lang'] = obj['@lang']
+                            o = rdflib.Literal(obj['@value'],**args)
+                        if object is not None and o != object:
+                            continue
+                        yield resourceid, rdflib.URIRef(pred), o, graphid
+
+def evalBGP(ctx, part=None, bgp=None):
```

Comment:
Duplicated?

Reply:
This code was not used, and as such was removed.